src/main/java/org/haffson/adventofcode/days/day06/Coordinate.java

```diff
@@ -0,0 +1,358 @@
+         * Is it claimed by more coordinates it should be marked with a ".".
+         *
+         * @return String to fill the coordinate system.
+         */
+        @Nonnull
+        private String givesCoordinateSystemInput(@Nonnull final List<Integer> nearestCoordinates) {
+            if (nearestCoordinates.size() > 1) {
+                return ".";
+            } else {
+                return String.valueOf(nearestCoordinates.get(0));
+            }
+        }
+
+        /**
+         * Helper method for day 6.
+         * Iterates over the coordinate system.
+         * Fills each entry with a coordinate id or "." when a spot is claimed by several coordinates
+         */
+        private void fillCoordinateSystem() {
+            Arrays.stream(coordinateSystem).forEach(arrayOfXCoordinates ->
+                    IntStream.range(0, arrayOfXCoordinates.length)
+                            .forEach(yCoordinate -> {
+                                int xCoordinate = Arrays.asList(coordinateSystem).indexOf(arrayOfXCoordinates);
+                                Coordinate toCompare = new Coordinate(xCoordinate, yCoordinate);
+
+                                List<Integer> nearestCoordinates = findNearestCoordinates(toCompare);
+                                coordinateSystem[xCoordinate][yCoordinate] = givesCoordinateSystemInput(nearestCoordinates);
+                            }));
+        }
+
+        /**
+         * Helper method for day 6.
+         * After a coordinate system is filled with coordinate ids it returns infinite coordinates.
+         * Coordinates who grow till infinity.
+         * To find those, this method irritates over all 4 borders and collects the coordinate ids to a Set.
+         *
+         * @return Set<String> with coordinate ids who grow till infinity.
+         */
+        @Nonnull
+        private Set<String> findInfiniteCoordinates() {
+            Set<String> infiniteCoordinates = Arrays.stream(coordinateSystem)
+                    .map(coordinates -> coordinates[0])
+                    .filter(coordinateSystem -> !Objects.equals(coordinateSystem, "."))
+                    .collect(Collectors.toSet());
+
+            Arrays.stream(coordinateSystem)
+                    .map(coordinates -> coordinates[coordinateSystem[0].length - 1])
+                    .filter(coordinateSystem -> !Objects.equals(coordinateSystem, "."))
+                    .forEach(infiniteCoordinates::add);
+
+            Arrays.stream(coordinateSystem[0])
+                    .filter(coordinateSystem -> !Objects.equals(coordinateSystem, "."))
+                    .forEach(infiniteCoordinates::add);
+
+            Arrays.stream(coordinateSystem[coordinateSystem.length - 1], 0, coordinateSystem[0].length)
+                    .filter(coordinateSystem -> !Objects.equals(coordinateSystem, "."))
+                    .forEach(infiniteCoordinates::add);
```

Comment:
Du kannst mehrere Streams auch mit `Stream.concat()` zusammenführen und dann einmalig am Ende collecten.

Reply:
Dadurch habe ich diese tolle Verschachtelung nun erhalten:
` final Stream<String> borderCoordinates = Stream.concat(
                    Stream.concat(
                            Stream.concat(
                                    topBorderCoordinates, bottomBorderCoordinates), leftBorderCoordinates), rightBorderCoordinates);
            return borderCoordinates.collect(Collectors.toSet());`

Gefällt mir ehrlich gesagt nur so semi-gut.

Auf Bealdung bin ich noch auf folgende syntax gestoßen:
`Stream<Integer> resultingStream = Stream.of(
      stream1, stream2, stream3, stream4)
      .flatMap(i -> i);

return resultingStream.collect(Collectors.toSet());`

Da würde mir die zweite Variante mehr zusagen, insofern diese keine versteckten Nachteile bereit hält.

Reply:
Ja, `concat()` akzeptiert leider nur 2 Parameter.


Das Konstrukt mit `flatMap()` ist auch die Empfehlung aus den API-Docs:
![Screenshot from 2022-08-03 13-02-27](https://user-images.githubusercontent.com/1383323/182593207-e6e8b453-e7f4-4fdf-82a9-e4627a145ea5.png)